### ISSUE TITLE
Is the documentation regarding options out of date?

### ISSUE THREAD

**Issue by jjs105** [tj/commander.js#1531]
Hi,

I have just started using commander and have been trying your examples (from the readme and example files) and not getting the expected results.

I think this is because the examples are out of date as follows (considering the options-common.js):

* It seems that the inclusion of boolean options as 'unassigned' in the result of the '.opts()' call doesn't happen, rather the properties simply do not exist
* After inspecting the code it seems that enabling the legacy option using '.storeOptionsAsProperties()' makes the example work as described

From the example, the following should occur:
```
$ pizza-options -d
{ debug: true, small: undefined, pizzaType: undefined }
```
However you actually get (unless setting legacy options to true):
```
$ pizza-options -d
{ debug: true }
```

TLDR; the new and legacy options objects act differently - this might be a bug or alternatively an documentation issue.

**Comment by shadowspawn** [comment]
Yes, the README is out of date and shows the old behaviour. Thanks.

The treatment of undefined options in `.opts()` does change depending on `. storeOptionsAsProperties()`. 

`.storeOptionsAsProperties()` was added in Commander v4.1, but until Commander v7.0 the default behaviour was to store the options as properties so the README was still correct. With 7.0 the README became out of date.

**Comment by shadowspawn** [comment]
Released in Commander v8.
